### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by TimothyMothra**
#1214 
Our Readme file hasn't been updated since before the repo consolidation.
Most of the examples and URLs were either out of date or specific to the BaseSDK.

## Changes
- Readme now provides a very brief overview of the repository, with links to other docs.
- New document "concepts.md" has concise descriptions and examples. links to MS Docs for detailed descriptions.
- Updated Contributing.md 

### Checklist
- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile locally using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -17,41 +22,16 @@ If you're interested in contributing, take a look at the general [contributer's
...
-Once you've configured the strong name verification, execute the ```runUnitTests.cmd``` script in the repository root.
-
-If the script fail with errors like unable to find path to Visual Studio Test runner, please edit the helper script to match you local installation of Visual Studio.
-
-You can also run the tests within Visual Studio using the test explorer. If test explorer is not showing all the tests, please make sure you have installed all updates to Visual Studio.
-
-You can remove the strong name verification exception by running this command as Administrator:
-
-    "%ProgramFiles(x86)%\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.7.1 Tools\sn.exe" -Vr ..\bin\Debug\Src\WindowsServer\WindowsServer.Net45.Tests\Microsoft.WindowsAzure.ServiceRuntime.dll
+Unit tests can be run in either the Visual Studio Test Exploror or via command line `dotnet test`.
```
Comment (eddynaka) [inline .github/CONTRIBUTING.md:34]: we have a typo here, Visual Studio Test Explorer

File: .docs/concepts.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Application Insights DotNet SDK Concepts
+This lists the high level concepts of our SDK and links to detailed guides to help you get started.
```
Comment (reyang) [inline .docs/concepts.md:2]: Minor: consider change `our` to a more specific word.

File: .docs/concepts.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+### Initialization
+You must initialize an instance of this object and populate it with your Instrumentation Key to identify your data.
+
+```C#
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+
+var configuration = new TelemetryConfiguration
+{
+    // Please use either an Instrumentation Key or a Connection String but not both.
```
Comment (reyang) [inline .docs/concepts.md:24]: Given the direction is to use Connection String, I think we should remove Instrumentation Key from the conceptual document.

File: .docs/concepts.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+var tc = new TelemetryClient(configuration);
+```
+
+### Using the TelemetryClient to send telemetry
+You can populate common context on the `TelemetryClient.context` property which will be automatically attached to each telemetry item sent. 
+You can also attach additional property data to each telemetry item sent. 
+The `TelemetryClient` also exposes a number of `Track` methods that can be used to send all telemetry types understood by the Application Insights service. Some example use cases are shown below.
+
+```C#
+tc.Context.User.Id = Environment.GetUserName(); // This is probably a bad idea from a PII perspective.
```
Comment (reyang) [inline .docs/concepts.md:37]: Probably replace this with a proper example.

File: .docs/concepts.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+tc.Context.Device.OperatingSystem = Environment.OSVersion.ToString();
+
+tc.TrackPageView("Form1");
+
+tc.TrackEvent("PurchaseOrderSubmitted", new Dictionary<string, string>() { {"CouponCode", "JULY2015" } }, new Dictionary<string, double>() { {"OrderTotal", 68.99 }, {"ItemsOrdered", 5} });
```
Comment (reyang) [inline .docs/concepts.md:42]: Change this to a better formatted code snippet instead of single line.

File: .docs/concepts.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	
+try
+{
+	...
```
Comment (reyang) [inline .docs/concepts.md:46]: Use 4 spaces instead of a tab.

File: .docs/concepts.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+}
+catch(Exception e)
+{
+	tc.TrackException(e);
+}
+``` 
+
+### TelemetryClient API
+The TelemetryClient provides several Track methods to collect different types of logs
```
Comment (reyang) [inline .docs/concepts.md:55]:
```suggestion
The `TelemetryClient` provides several Track methods to collect different types of telemetry data.
```

File: .docs/concepts.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+Please review our [Api summary for custom events and metrics](https://docs.microsoft.com/azure/azure-monitor/app/api-custom-events-metrics).
+
```
Comment (reyang) [inline .docs/concepts.md:58]: Nit: remove extra empty lines here.

File: .docs/concepts.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+catch(Exception e)
+{
+	tc.TrackException(e);
+}
+``` 
+
+### TelemetryClient API
+The TelemetryClient provides several Track methods to collect different types of logs
+
+Please review our [Api summary for custom events and metrics](https://docs.microsoft.com/azure/azure-monitor/app/api-custom-events-metrics).
```
Comment (reyang) [inline .docs/concepts.md:57]:
```suggestion
Please review the [API summary for custom events and metrics](https://docs.microsoft.com/azure/azure-monitor/app/api-custom-events-metrics).
```

File: .docs/concepts.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+
+
+## Telemetry Channels
+Telemetry channels are responsible for buffering telemetry items and sending them to the Application Insights service, where they're stored for querying and analysis.
```
Comment (reyang) [inline .docs/concepts.md:62]: Not necessary to have `buffering`?

File: .docs/concepts.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+The .NET and .NET Core versions of the SDKs provide two built-in telemetry channels::
```
Comment (reyang) [inline .docs/concepts.md:64]:
```suggestion
The .NET and .NET Core versions of the SDKs provide two built-in telemetry channels:
```

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -107,18 +87,21 @@ Edit the helper scripts to change between 'Release' and 'Debug' as per your buil
 Its is important to note that functional tests do not trigger product code build, so explicit build of product code is required before running functional tests.
 A typical work flow would be make-produce-change followed by build-product followed by build-functest-solution and finally run-func-tests. (This helpers scripts does this.)
 
-## Known issues/workarounds with running functional tests.
+### Known issues/workarounds with running functional tests.
 
 If any tests fail, please retry first to see if it helps. If not, try one of the known issues below. 
 
-Tests fail with error like "It was not possible to find any compatible framework version The specified framework 'Microsoft.NETCore.App', version '1.0.4' was not found"
+If these don't help, please open an issue in Github describing the problem.
```
Comment (reyang) [inline .github/CONTRIBUTING.md:94]: Minor, it'll be nice to link to the actual issue template.

File: Readme.md [hunk 13]
```diff
@@ -1,13 +1,40 @@
...
+# Application Insights for .NET
+This is the .NET SDK for [Azure Monitor](https://docs.microsoft.com/azure/azure-monitor/overview) & [Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/app-insights-overview).
+
+
+
+## Getting Started
+Please review our How-to guides to review which packages are appropriate for your project:
+- [.NET Console](https://docs.microsoft.com/azure/azure-monitor/app/console)
+- [ASP.NET](https://docs.microsoft.com/azure/azure-monitor/app/asp-net)
+- [ASP.NET Core](https://docs.microsoft.com/azure/azure-monitor/app/asp-net-core)
```
Comment (reyang) [inline Readme.md:10]: Curious: Is there a .NET Core Console?

File: .docs/concepts.md (hunk 6 above)
Comment (TimothyMothra) [reply .docs/concepts.md:46]: How did you even see that? o_O

File: .docs/concepts.md (hunk 10 above)
Comment (TimothyMothra) [reply .docs/concepts.md:62]: I didn't write any of these statements. That's an exact quote from our public doc. Do you think it should say something else here?

File: Readme.md (hunk 13 above)
Comment (TimothyMothra) [reply Readme.md:10]: Visual Studio does have a template for .NET Core Console Apps, but there's nothing special about them. they look just like other console projects.
We don't have a public doc for these project types, but I'm not certain we need one. :)

Comment (reyang) [reply Readme.md:10]: Maybe we just replace `.NET Console` with `Console`? The document link also talked about `app/console` without specifying whether it is .NET Framework or .NET Core. In case the document is missing .NET Core part, it should be a bug on Azure Docs.

File: .docs/concepts.md (hunk 3 above)
Comment (TimothyMothra) [reply .docs/concepts.md:24]: I'm trying to ignore my bias here because I implemented Connection Strings. :)

The UX still displays the Instrumentation Key on the Overview blade. Since it's a very simple way to get started with our SDK I think it's still a valid example for getting started. 
I don't feel strongly about it, but most of our official docs still use the ikey example for getting started. 

Maybe this should be a larger refactoring effort for the docs team to review all public docs and official repos?

File: Readme.md (hunk 13 above)
Comment (TimothyMothra) [reply Readme.md:10]: Ah, I see what you're saying. Yes that makes more sense.

File: .docs/concepts.md (hunk 3 above)
Comment (reyang) [reply .docs/concepts.md:24]: We can start here. Need help from @ank3it to drive other docs to catch up.

The general principle is that we should only give *one recommended path* in `conceptual documents`. It is okay to list _all possible ways_ in `reference document`.

Comment (TimothyMothra) [reply .docs/concepts.md:24]: Thanks for clarifying conceptual vs reference. I'll keep that in mind going forward!

File: Readme.md [hunk 14]
```diff
@@ -1,13 +1,40 @@
-# Please pardon our progress
-We are currently in the process of consolidating our repos.
-This Readme will be updated as we make significant changes.
+# Application Insights for .NET
```
Comment (ank3it) [inline Readme.md:1]: can we rephrase this to "Application Insights for .NET Apps"

File: Readme.md [hunk 15]
```diff
@@ -1,13 +1,40 @@
...
+This is the .NET SDK for [Azure Monitor](https://docs.microsoft.com/azure/azure-monitor/overview) & [Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/app-insights-overview).
```
Comment (ank3it) [inline Readme.md:2]: Rephrase to 
"This is the .NET SDK for sending data to [Azure Monitor]"

File: .docs/concepts.md (hunk 10 above)
Comment (mrbullwinkle) [reply .docs/concepts.md:60]: We can kill buffering in the public docs as well if needed. A lot of the core terminology used in the public docs predates all of us working on the product so if there is something that doesn't make sense, definitely let me know so we can clean it up.

Comment (reyang) [reply .docs/concepts.md:63]: I wonder if this would help to clarify "Telemetry channels are responsible for sending the telemetry data to the designated place. Optional features can be provided by the telemetry channels, for example, buffering the data and sending in them in batches, persisting the data to a local storage in case of transmission failure (e.g. network outage), traffic shaping and retry mechanisms".

Comment (TimothyMothra) [reply .docs/concepts.md:63]: How about:
"Telemetry channels are responsible for sending telemetry data. Optional features include buffering and batching data, local data storage in case of transmission failure, and retry mechanisms."

Comment (cijothomas) [reply .docs/concepts.md:63]: Why dont we just link to public doc here? for every topic where there is a public doc, i'd suggest to simply link it here and not write anything more.

Comment (TimothyMothra) [reply .docs/concepts.md:63]: @cijothomas, i've given every section an intro statement, copied from the public doc and also a link to that public doc. 
The purpose is if someone knew nothing about the sdk, they could navigate high-level concepts all in one place.
Similar to this: https://github.com/microsoft/ApplicationInsights-Home/issues/256
